sbi/neural_nets/density_estimators/nflows_flow.py

```diff
@@ -135,12 +135,14 @@ def sample(self, sample_shape: Shape, condition: Tensor) -> Tensor:
         num_samples = torch.Size(sample_shape).numel()
 
         samples = self.net.sample(num_samples, context=condition)
-
-        return samples.reshape((
-            *sample_shape,
-            condition_batch_dim,
-            -1,
-        ))
+        samples = samples.transpose(0, 1)
+        return samples.reshape(
+            (
+                *sample_shape,
+                condition_batch_dim,
+            )
+            + self.input_shape
```

Comment:
thanks! Would
```python
(*sample_shape, condition_batch_dim, *self.input_shape)
```
also work? It's just a bit cleaner IMO.

Reply:
Jup, fixed.